File: Data Structures & Algorithms/duplicate-integer/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        let mySet = new Set()

        for (let i = 0; i < nums.length; i++) {
            if (!mySet[nums[i]]) {
                mySet.add(nums[i])
            }
        }
        if (mySet.size != nums.length) {
            return true
        } else {
            return false
        }
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        n = len(nums)
        L = 0
        R = n - 1

        if nums[L] <= nums[R]:
            return nums[L]

        while L <= R:

            mid = L + (R - L) // 2

            if nums[mid] < nums[(mid - 1) % n]:
                return nums[mid]
            
            elif nums[mid] >= nums[0]:
                L = mid + 1
            
            else:
                R = mid - 1
